### ISSUE TITLE
Read Segment Anything Model (SAM) paper

### ISSUE THREAD

**Issue by JHuardC** [JHuardC/map-text-detection#1]
[paper](https://arxiv.org/pdf/2304.02643)

**Comment by JHuardC** [comment]
##  Dataset generation
Borrow from section 4 as inspiration for how to develop the new mask-text dataset through a bootstrapping process:
- Take point data from the OS text search product
- Apply zero shot transfer using SAM to obtain text masks
- Clean up masks and use as a gold standard dataset
- Train text-spotters and/or parsers on this dataset
- Expand the dataset through **semi-automatic** labelling: 
  - Apply the trained model to sample areas of the 6-inch map
  - Assess and gather output masks
  - Clean up partial masks
  - Add correct/cleansed mask the gold standard train/validate dataset 
  - Retrain the model
 - Once the required accuracy/coverage has been reached on the gold-standard train/validate dataset apply the model to the full dataset

## Useful excerpts
*When scaled and trained with
abundant text corpora from the web, these models’ zero and
few-shot performance compares surprisingly well to (even
matching in some cases) fine-tuned models [10, 21]. Empir
ical trends show this behavior improving with model scale,
dataset size, and total training compute [56, 10, 21, 51].* - Page 1

*While SAM performs well in general, it is
not perfect. It can miss fine structures, hallucinates small
disconnected components at times, and does not produce
boundaries as crisply as more computationally intensive
methods that “zoom-in”, e.g. [18].* - Page 12

## SAM augmentation
Apply adaptors to image encoder and mask decoder to improve text detection on full automatic segmentation?

## Data Augmentation
Introduce map noise to improve model by improving background detection: Golnaz Ghiasi, Yin Cui, Aravind Srinivas, Rui Qian, Tsung-Yi Lin,
Ekin DCubuk, QuocVLe,andBarretZoph. Simplecopy-paste is a
strong data augmentation method for instance segmentation. CVPR,
2021

## Further reading
Jimmy Lei Ba, Jamie Ryan Kiros, and Geoffrey E Hinton. Layer
normalization. arXiv:1607.06450, 2016.

Hangbo Bao, Li Dong, and Furu Wei. BEiT: BERT pre-training of
image transformers. arXiv:2106.08254, 2021.

Rishi Bommasani, Drew A Hudson, Ehsan Adeli, Russ Altman,
Simran Arora, Sydney von Arx, Michael S Bernstein, Jeannette
Bohg, Antoine Bosselut, Emma Brunskill, et al. On the opportu
nities and risks of foundation models. arXiv:2108.07258, 2021.

Nicolas Carion, Francisco Massa, Gabriel Synnaeve, Nicolas
Usunier, Alexander Kirillov, and Sergey Zagoruyko. End-to-end
object detection with Transformers. ECCV, 2020.

Alexey Dosovitskiy, Lucas Beyer, Alexander Kolesnikov, Dirk
Weissenborn, Xiaohua Zhai, Thomas Unterthiner, Mostafa De
hghani, Matthias Minderer, Georg Heigold, Sylvain Gelly, Jakob
Uszkoreit, and Neil Houlsby. An image is worth 16x16 words:
Transformers for image recognition at scale. ICLR, 2021.

Marco Forte, Brian Price, Scott Cohen, Ning Xu, and Franc¸ois
Piti´e.
Getting to 99% accuracy in interactive segmentation.
arXiv:2003.07932, 2020.

Golnaz Ghiasi, Yin Cui, Aravind Srinivas, Rui Qian, Tsung-Yi Lin,
Ekin DCubuk, QuocVLe,andBarretZoph. Simplecopy-paste is a
strong data augmentation method for instance segmentation. CVPR,
2021.

Kaiming He, Xinlei Chen, Saining Xie, Yanghao Li, Piotr Doll´ar,
and Ross Girshick. Masked autoencoders are scalable vision learn
ers. CVPR, 2022.

Bowen Cheng, Ishan Misra, Alexander G Schwing, Alexander Kir
illov, and Rohit Girdhar. Masked-attention mask transformer for
universal image segmentation. CVPR, 2022.

XiChen, Zhiyan Zhao, Yilei Zhang, Manni Duan, Donglian Qi, and
Hengshuang Zhao. FocalClick: towards practical interactive image
segmentation. CVPR, 2022. 8, 9, 12, 19

### Maybe
Tom Brown, Benjamin Mann, Nick Ryder, Melanie Subbiah,
Jared D Kaplan, Prafulla Dhariwal, Arvind Neelakantan, Pranav
Shyam, Girish Sastry, Amanda Askell, Sandhini Agarwal, Ariel
Herbert-Voss, Gretchen Krueger, Tom Henighan, Rewon Child,
Aditya Ramesh, Daniel Ziegler, Jeffrey Wu, Clemens Winter, Chris
Hesse, Mark Chen, Eric Sigler, Mateusz Litwin, Scott Gray, Ben
jamin Chess, Jack Clark, Christopher Berner, Sam McCandlish,
Alec Radford, Ilya Sutskever, and Dario Amodei. Language models
are few-shot learners. NeurIPS, 2020.

**Comment by JHuardC** [comment]
Approximately 35s to make 1 mask
